packages/messaging/src/api.ts

```diff
@@ -43,6 +45,22 @@ import { _setDeliveryMetricsExportedToBigQueryEnabled } from './api/setDeliveryM
  * @public
  */
 export function getMessagingInWindow(app: FirebaseApp = getApp()): Messaging {
+  // Conscious decision to make this async check non-blocking during the messaging instance
+  // initialization phase for performance consideration. An error would be thrown latter for
+  // developer's information. Developers can then choose to import and call `isSupported` for
+  // special handling.
+  isWindowSupported()
+    .then(isSupported => {
+      if (!isSupported) {
+        throw ERROR_FACTORY.create(ErrorCode.UNSUPPORTED_BROWSER);
+      }
+    })
+    .catch((e: Error) => {
+      if (e.message.includes(ErrorCode.UNSUPPORTED_BROWSER)) {
+        throw e;
+      }
+      throw ERROR_FACTORY.create(ErrorCode.INDEXED_DB_UNSUPPORTED);
+    });
```

Comment:
We can just use the error callback of `then`, which is much simpler.

Reply:
Done.